### PR TITLE
INT B 18889 remove prime ability to update destination address for approved shipment outside correct endpoint

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

When a shipment is in an `APPROVED` status, we do not want the Prime to have the freedom to update the destination address of a shipment without the TOO approving the change first. 

Right now there are two endpoints where the destination address can be updated without the respective logic checks that are in the `updateShipmentDestinationAddress` endpoint. This PR does the following:
- when calling the `updateMTOShipment` and `updateMTOShipmentAddress` endpoints, implemented logic to check the shipment status and if the shipment's destination address is provided in the `payload`
- handled error return to inform the user to use the `updateMTOShipmentDestinationAddress` endpoint instead when trying to update the destination address
- updated tests to reflect changes in logic

### How to test

1. Access MM as prime
2. Find a move with an approved shipment
3. Try to edit the destination address for the shipment by clicking the `Edit` and submitting the form
4. You should get back an error

## Screenshots
Screenshot 2024-03-14 at 2 42 46 PM

### CODE REVIEW

**Review by traskowskycaci — APPROVED** [review]

File: pkg/handlers/primeapi/mto_shipment.go [hunk 1]
```diff
@@ -180,6 +180,14 @@ func (h UpdateMTOShipmentHandler) Handle(params mtoshipmentops.UpdateMTOShipment
 					payloads.ClientError(handlers.NotFoundMessage, err.Error(), h.GetTraceIDFromRequest(params.HTTPRequest))), err
 			}
 
+			if dbShipment.Status == models.MTOShipmentStatusApproved &&
+				(params.Body.DestinationAddress.City != nil ||
+					params.Body.DestinationAddress.State != nil ||
+					params.Body.DestinationAddress.PostalCode != nil) {
+				return mtoshipmentops.NewUpdateMTOShipmentUnprocessableEntity().WithPayload(payloads.ValidationError(
+					"This shipment is approved, please use the updateShipmentDestinationAddress endpoint to update the destination address", h.GetTraceIDFromRequest(params.HTTPRequest), nil)), err
+			}
```
Comment (cameroncaci) [inline pkg/handlers/primeapi/mto_shipment.go:189]: Could we update this error response? As it checks for more than if the shipment is just approved

File: pkg/handlers/primeapi/mto_shipment_address.go [hunk 2]
```diff
@@ -30,6 +32,18 @@ func (h UpdateMTOShipmentAddressHandler) Handle(params mtoshipmentops.UpdateMTOS
...
+			dbShipment, err := mtoshipment.FindShipment(appCtx, mtoShipmentID, "DestinationAddress")
+			if err != nil {
+				return mtoshipmentops.NewUpdateMTOShipmentAddressNotFound().WithPayload(
+					payloads.ClientError(handlers.NotFoundMessage, err.Error(), h.GetTraceIDFromRequest(params.HTTPRequest))), err
+			}
+
+			if dbShipment.Status == models.MTOShipmentStatusApproved &&
+				(*dbShipment.DestinationAddressID == addressID) {
+				return mtoshipmentops.NewUpdateMTOShipmentAddressUnprocessableEntity().WithPayload(payloads.ValidationError(
+					"This shipment is approved, please use the updateShipmentDestinationAddress endpoint to update the destination address of an approved shipment", h.GetTraceIDFromRequest(params.HTTPRequest), nil)), err
```
Comment (cameroncaci) [inline pkg/handlers/primeapi/mto_shipment_address.go:44]: Same right here

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Updating these here will help us when debugging or help the Prime when using our API

**Comment by cameroncaci** [comment]
image

`Update Shipment Destination Address` should be functional on an approved shipment right?

**Comment by danieljordan-caci** [comment]
> image
> 
> Should `Udate Shipment Destination Address`be functional on an approved shipment?

Yessir

File: pkg/handlers/primeapi/mto_shipment.go (hunk 1 above)
Comment (danieljordan-caci) [reply pkg/handlers/primeapi/mto_shipment.go:189]: I would say this error response is appropriate. 
It's checking if the shipment status is approved, which `This shipment is approved` covers
And also checking is the dest address is in the payload, which `please use the updateShipmentDestinationAddress endpoint to update the destination address` covers

I'm open to other suggestions though.

Comment (cameroncaci) [reply pkg/handlers/primeapi/mto_shipment.go:189]: Thought about it and yeah that works. No other code changes necessary, lgtm!

**Pull request merged**